pkg/utils/yaml.go

```diff
@@ -26,56 +26,81 @@ var (
 func LoadYAML(filePath string) ([]*IacDocument, error) {
 	iacDocumentList := make([]*IacDocument, 0)
 
-	// First pass determines line number data
-	{ // Limit the scope for Close()
-		file, err := os.Open(filePath)
-		if err != nil {
-			return iacDocumentList, err
-		}
-		defer file.Close()
-
-		scanner := bufio.NewScanner(file)
-		startLineNumber := 1
-		currentLineNumber := 1
-		for scanner.Scan() {
-			if strings.HasPrefix(scanner.Text(), "---") {
-				// We've found the end-of-directives marker, so record results for the current document
-				iacDocumentList = append(iacDocumentList, &IacDocument{
-					Type:      YAMLDoc,
-					StartLine: startLineNumber,
-					EndLine:   currentLineNumber,
-					FilePath:  filePath,
-				})
-				startLineNumber = currentLineNumber + 1
-			}
-			currentLineNumber++
-		}
+	file, err := os.Open(filePath)
+	if err != nil {
+		return iacDocumentList, err
+	}
+	defer file.Close()
 
-		// Add the very last entry
-		iacDocumentList = append(iacDocumentList, &IacDocument{
-			Type:      YAMLDoc,
-			StartLine: startLineNumber,
-			EndLine:   currentLineNumber,
-			FilePath:  filePath,
-		})
+	fileBytes, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return iacDocumentList, err
+	}
 
-		if err = scanner.Err(); err != nil {
-			return iacDocumentList, err
+	return getIacDocumentList(bufio.NewScanner(file), fileBytes, filePath)
+}
+
+// LoadYAMLString loads a YAML String. Can return one or more IaC Documents.
+// Besides reading in file data, its main purpose is to determine and store line number and filename metadata
+func LoadYAMLString(data, absFilePath string) ([]*IacDocument, error) {
+	return getIacDocumentList(bufio.NewScanner(strings.NewReader(data)), []byte(data), absFilePath)
+}
+
+// ReadYamlFile reads a yaml file and load content in a map[string]interface{} type
+func ReadYamlFile(path string) (map[string]interface{}, error) {
+	dat, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	var output map[string]interface{}
+	err = yaml.Unmarshal(dat, &output)
+	if err != nil {
+		return nil, err
+	}
+	return output, nil
+}
+
+// getIacDocumentList provides one or more IaC Documents.
+// Besides reading in file data, its main purpose is to determine and store line number and filename metadata
+func getIacDocumentList(scanner *bufio.Scanner, bytearray []byte, filePath string) ([]*IacDocument, error) {
```

Comment:
Would it be better if the function name some how mentions YAML, as yaml documents are being created here?

Reply:
technically we're passing a scanner object as input